pkg/secretless/plugin/sharedobj/mock/plugins.go

```diff
@@ -121,3 +124,72 @@ func GetExternalPlugins(
 func NewLogger() log_api.Logger {
 	return log.New(true)
 }
+
+// RawPlugin implements the rawPlugin interface
+type RawPlugin struct {
+	PluginAPIVersion string
+	PluginType       string
+	PluginID         string
+}
+
+func (r RawPlugin) pluginInfo() map[string]string {
+	return map[string]string{
+		"pluginAPIVersion": r.PluginAPIVersion,
+		"type":             r.PluginType,
+		"id":               r.PluginID,
+	}
+}
+
+func (r RawPlugin) httpPlugin() http.Plugin {
+	return NewHTTPPlugin(r.PluginID)
+}
+
+func (r RawPlugin) tcpPlugin() tcp.Plugin {
+	return NewTCPPlugin(r.PluginID)
+}
+
+// Lookup returns a go_plugin.Symbol for a given symbol name string. A go_plugin.Symbol
+// is an empty interface, and can be instantiated with a function or a struct.
+// This method is intended to mimic the Lookup method for a standard Go plugin.
+func (r RawPlugin) Lookup(symbolName string) (go_plugin.Symbol, error) {
+	switch symbolName {
+	case "PluginInfo":
+		return r.pluginInfo, nil
+	case "GetHTTPPlugin":
+		return r.httpPlugin, nil
+	case "GetTCPPlugin":
+		return r.tcpPlugin, nil
+	}
+	return nil, fmt.Errorf("unknown symbolName %s", symbolName)
+}
+
+// RawPlugins instantiates a map of RawPlugins for testing. Values of entries
+// in this map should reflect values returned by HTTPExternalPluginsByID()
+// and TCPExternalPluginsByID() above.
+var RawPlugins = map[string]RawPlugin{
+	"http1": RawPlugin{
+		PluginAPIVersion: "0.1.0",
```

Comment:
This string can be a constant since it's repeated a bunch

Reply:
Fixed.